### PR TITLE
fix(hazard_status_converter): check current operation mode

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Ignore errors during initialization.
https://github.com/autowarefoundation/autoware.universe/blob/main/system/system_error_monitor/src/system_error_monitor_core.cpp#L177-L202

## Tests performed

1. Disable emergency_handler and system_error_monitor
2. Launch psim
3. Launch following command in another terminal
    `ros2 launch hazard_status_converter hazard_status_converter.launch.xml`
4. Launch following command in another terminal
    `ros2 launch diagnostic_graph_aggregator example-main.launch.xml`
5. Change operation mode and check if SPF only in that mode

## Effects on system behavior

Fix to be the same behavior of previous hazard_status

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/hazard_status_converter/src/converter.cpp [hunk 1]
```diff
@@ -148,7 +169,32 @@ void Converter::on_graph(const DiagnosticGraph::ConstSharedPtr msg)
     return HazardStatus::NO_FAULT;
   };
 
-  HazardStatusStamped hazard = convert_hazard_diagnostics(*msg);
+  const auto is_ignore = [this]() {
+    if (mode_) {
+      if (mode_->mode == OperationMode::AUTONOMOUS) {
```
Comment (TomohitoAndo) [inline system/hazard_status_converter/src/converter.cpp:174]: [Q] In [system_error_monitor](https://github.com/autowarefoundation/autoware.universe/blob/dfec62ac277f893fd6276e3c6ffd9c7d4f475225/system/system_error_monitor/src/system_error_monitor_core.cpp#L190) `GateMode` is used, but `OperationMode` is used here. What is the difference of these?

**Review by TomohitoAndo — APPROVED** [review]
Although I have added one question, the code looks good to me!

File: system/hazard_status_converter/src/converter.cpp (hunk 1 above)
Comment (isamu-takagi) [reply system/hazard_status_converter/src/converter.cpp:174]: It's almost the same. The operation mode is more abstract.
https://autowarefoundation.github.io/autoware.universe/main/system/default_ad_api/document/operation-mode/#compatibility

Comment (isamu-takagi) [reply system/hazard_status_converter/src/converter.cpp]: FIxed for stop mode.
https://github.com/autowarefoundation/autoware.universe/pull/6733/commits/1e561c2994bf17589288e3519563b252525e02f5

Comment (TomohitoAndo) [reply system/hazard_status_converter/src/converter.cpp]: Thank you!

**Pull request merged**